ext/opcache/jit/zend_jit_ir.c

```diff
@@ -11198,26 +11156,20 @@ static int zend_jit_leave_func(zend_jit_ctx         *jit,
 		 && (JIT_G(current_frame) && !TRACE_FRAME_IS_UNKNOWN_RETURN(JIT_G(current_frame)))) {
 			zend_jit_reset_last_valid_opline(jit);
 		} else {
-			if (GCC_GLOBAL_REGS) {
-				/* We add extra RLOAD and RSTORE to make fusion for persistent register
-				 *     mov (%FP), %IP
-				 *     add $0x1c, %IP
-				 * The naive (commented) code leads to extra register allocation and move.
-				 *     mov (%FP), %tmp
-				 *     add $0x1c, %tmp
-				 *     mov %tmp, %FP
-				 */
+			/* We add extra RLOAD and RSTORE to make fusion for persistent register
+			 *     mov (%FP), %IP
+			 *     add $0x1c, %IP
+			 * The naive (commented) code leads to extra register allocation and move.
+			 *     mov (%FP), %tmp
+			 *     add $0x1c, %tmp
+			 *     mov %tmp, %FP
+			 */
 #if 0
-				jit_STORE_IP(jit, ir_ADD_OFFSET(ir_LOAD_A(jit_EX(opline)), sizeof(zend_op)));
+			jit_STORE_IP(jit, ir_ADD_OFFSET(ir_LOAD_A(jit_EX(opline)), sizeof(zend_op)));
 #else
-				jit_STORE_IP(jit, ir_LOAD_A(jit_EX(opline)));
-				jit_STORE_IP(jit, ir_ADD_OFFSET(jit_IP(jit), sizeof(zend_op)));
+			jit_STORE_IP(jit, ir_LOAD_A(jit_EX(opline)));
+			jit_STORE_IP(jit, ir_ADD_OFFSET(jit_IP(jit), sizeof(zend_op)));
```

Comment:
These stores are not necessary for CALL VM.

```
	movq (%r14), %r15      => movq (%r14), %rax
	addq $0x20, %r15       => addq #0x20, %rax
	movq %r15, %rax        => nop
	btsq $0x3f, %rax         
	addq $0x38, %rsp
	popq %r15
	popq %r14
	popq %r13
	popq %r12
	popq %rbp
	popq %rbx
	retq
```

Reply:
Looked into this but it's difficult to avoid, as the store is in `zend_jit_leave_func()`, but the `movq %r15, %rax; btsq $0x3f, %rax` is from `zend_jit_trace_return()`.

Do you have a suggestion how to address this?

If that's ok for you, I would like to merge this PR (waiting a few days for other reviews) and move to the next step of https://github.com/php/php-src/pull/17849 (tail calling), as this may change a few things again.

After that I will come back to this comment as well as https://github.com/php/php-src/pull/17952#discussion_r2021597301.